### PR TITLE
Testing out bun with NX/TSC/Vite

### PR DESCRIPTION
**Pull request by bhouston**
Hacked in bun into the nx workflows and CI and docker build.  Seems to produce a 30% speed increase.

Unfortunately the docker build is failing currently, probably because of nx incompatibility with bun.

I've mentioned the issue as well as proposing that NX support bun here:
* https://github.com/nrwl/nx/issues/21075
* https://github.com/nrwl/nx/discussions/11048#discussioncomment-8080713

### CODE REVIEW

File: apps/react-app/project.json [hunk 1]
```diff
@@ -4,12 +4,12 @@
...
     "watch": {
       "options": {
-        "commands": ["vite"]
+        "commands": ["bunx vite"]
       }
     },
     "build": {
       "options": {
-        "commands": ["tsc", "vite build"]
+        "commands": ["bunx tsc", "bunx vite build"]
```
Comment (Jordan-Hall) [inline apps/react-app/project.json]: using bunx vite will basically run vite in node and not bun. Same with TSC here. You need force bun with --bun
`
``suggestion
        "commands": ["bunx --bun tsc", "bunx --bun vite build"]
```

bun also has bulit in support for typescript compling so you dont need to run TSC. Whats the reasoning for it?

File: apps/react-app/project.json [hunk 2]
```diff
@@ -4,12 +4,12 @@
   "targets": {
     "watch": {
       "options": {
-        "commands": ["vite"]
+        "commands": ["bunx vite"]
```
Comment (Jordan-Hall) [inline apps/react-app/project.json]:
```suggestion
        "commands": ["bunx --bun vite"]
```

File: apps/koa-server/project.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   "targets": {
     "watch": {
       "options": {
-        "commands": ["tsc --watch", "nodemon dist/server.js"]
+        "commands": ["bunx tsc --watch", "bunx nodemon dist/server.js"]
```
Comment (Jordan-Hall) [inline apps/koa-server/project.json]: Read comment below i think you could change this to the following

```suggestion
        "commands": ["bun run --watch server.ts"]
```

**Review by Jordan-Hall — CHANGES_REQUESTED** [review]

File: apps/koa-server/project.json (hunk 3 above)
Comment (bhouston) [reply apps/koa-server/project.json]: Done!  Thx!

File: apps/react-app/project.json (hunk 2 above)
Comment (bhouston) [reply apps/react-app/project.json]: I've adopted this!  Thanks!

File: apps/react-app/project.json (hunk 1 above)
Comment (bhouston) [reply apps/react-app/project.json]: done!  thx!

Comment (bhouston) [reply apps/react-app/project.json]: @Jordan-Hall regarding:

"bun also has bulit in support for typescript compling so you dont need to run TSC. Whats the reasoning for it?"

I want to publish reusable libraries for others from some of my projects.  Thus I want to transpire to *.js that is non-bundled, non-minified for easy reuse, tree shaking, etc.

I guess you'd recommend that I use https://bun.sh/docs/bundler?  I guess that can be used for libraries and not just end user React apps?

Comment (Jordan-Hall) [reply apps/react-app/project.json]: Correct working on getting this to work on nx-bun but you can use cli for it


bun build ./index.tsx --outdir ./out

**Comment by Jordan-Hall** [comment]
Nife to change to force bun.  Bun as a package manager is supper quick. But they also limit to approved  postinstall via bun install. It can get 2 to 4x the speed. But this is what I was referring to on NX. You can use bun as PM people seeing instant result which is strange still

File: Dockerfile [hunk 4]
```diff
@@ -1,13 +1,12 @@
-FROM node:21-alpine
+FROM imbios/bun-node:latest-21-slim
```
Comment (Jordan-Hall) [inline Dockerfile]:
```suggestion
FROM oven/bun:1
```

Now you are running it in actual bun (vite etc) you shouldn't need node here anymore

**Pull request merged**